src/app/payments/update-pending-payments.ts

```diff
@@ -187,15 +188,27 @@ const updatePendingPayment = async ({
           { success: false, id: paymentHash, payment: pendingPayment },
           "payment has failed. reverting transaction",
         )
-
-        const voided = await ledgerService.revertLightningPayment({
-          journalId: pendingPayment.journalId,
-          paymentHash,
-        })
-        if (voided instanceof Error) {
-          const error = `error voiding payment entry`
-          logger.fatal({ success: false, result: lnPaymentLookup }, error)
-          return voided
+        if (paymentFlow.senderWalletCurrency === WalletCurrency.Btc) {
+          const voided = await ledgerService.revertLightningPayment({
+            journalId: pendingPayment.journalId,
+            paymentHash,
+          })
+          if (voided instanceof Error) {
+            const error = `error voiding payment entry`
+            logger.fatal({ success: false, result: lnPaymentLookup }, error)
+            return voided
+          }
+        } else {
+          const reimbursed = await Wallets.reimburseFailedUsdPayment({
+            journalId: pendingPayment.journalId,
+            paymentFlow,
+            accountId: wallet.accountId,
+          })
+          if (reimbursed instanceof Error) {
+            const error = `error reimbursing usd payment entry`
+            logger.fatal({ success: false, result: lnPaymentLookup }, error)
```

Comment:
as we rely less and less on logs and more and more on opentelemetry for alert, I'm wondering if we should not have an event here in addition, or instead?

the current `Error` being returned here would not triggered back an alert on honeycomb.

ie: `InvalidAccountForWalletIdError`, `CouldNotFindWalletForWalletCurrencyError`, `CouldNotListWalletsFromAccountIdError` are not `ErrorLevel.Critical` (nor they should be in other contexts)

Reply:
Good point, done: https://github.com/GaloyMoney/galoy/pull/1535/commits/750bdbeaac5f6afa95c41013e310c4f4e5e920b9